clients/drcachesim/common/trace_entry.h

```diff
@@ -163,6 +163,49 @@ typedef enum {
      */
     TRACE_TYPE_INSTR_SYSENTER,
 
+    /** AArch64 specific prefetch trace entries.
+     *  Relative order of these enum values is relevant to the handling of AArch64
+     * prefetch instructions.
+     */
+    TRACE_TYPE_PREFETCH_PLDL1KEEP, /**< An AArch64 load prefetch to level 1 of the cache.
```

Comment:
I renamed the enum strings to be architecture agnostic. 

> If this is the same as TRACE_TYPE_PREFETCH_READ let's just use that.

AArch64 prefetches have more variations than x86 or ARM -- {read, instr, write} x {L1, L2, L3} x {temporal, non-temporal}. So we'll need to add new enum values.

> I see that some of the existing prefetches do mention ARM or x86, but IMHO we should try to remove that too.

Do you mean we should remove the existing enum values and reuse these new ones for all architectures?

Reply:
> > If this is the same as TRACE_TYPE_PREFETCH_READ let's just use that.
> 
> AArch64 prefetches have more variations than x86 or ARM -- {read, instr, write} x {L1, L2, L3} x {temporal, non-temporal}. So we'll need to add new enum values.

Right, but only new ones for the ones that don't already exist.
 
> > I see that some of the existing prefetches do mention ARM or x86, but IMHO we should try to remove that too.
> 
> Do you mean we should remove the existing enum values and reuse these new ones for all architectures?

Not remove, but change the doc description and add an arch-agnostic alias enum value.  Could be a separate PR (but this would be chained onto it if the aarch64 types overlap).

Reply:
I'm not sure if we can share the same set of enums across all architectures.

> Right, but only new ones for the ones that don't already exist.

The existing enums describe prefetches with a limited set of properties (only the target cache level or operation type). For AArch64, we need enums that describe all three properties: operation type, target cache level, locality (streaming/temporal). If I understand correctly, we cannot use any of the existing enums for AArch64.

Maybe we can use some of the new enums for x86/ARM. ARM's `TRACE_TYPE_PREFETCH_READ` PLD instruction can perhaps be deprecated in favour of `TRACE_TYPE_PREFETCH_READ_L1`. But for some others, the mapping is not straight-forward - like `TRACE_TYPE_PREFETCHT2` doesn't specify whether there's intent to write or not.

Reply:
Agreed, if an existing one does not line up we would keep it separate and make a new one.  But if they are identical semantics, IMHO sharing is the best thing, as it will simplify trace analysis and simulation tools.

Reply:
I added some changes to address this comment. It seems complicated to get an architecture independent set of trace types for prefetch instructions.

About 32-bit ARM prefetch instructions (PLD, PLDW and PLI), ARM ISA does not specify the target cache level[1]. I did find some references that mention a target cache level[2][3], but it seems implementation defined. As we don't know the target cache level, the existing ARM-specific prefetch trace types don't seem to map to any of the new generic ones.

About x86, the PREFETCHTX instrs map to TRACE_TYPE_PREFETCH_READ_LX trace types. (Based on [4], these prefetches do _not_ anticipate write.)
However, PREFETCHW (which anticipates write) can prefetch into L1 _or_ L2[5], so that also doesn't map directly to any one of the new trace type. 

[1]: https://static.docs.arm.com/100076/0100/arm_instruction_set_reference_guide_100076_0100_00_en.pdf C2-272
[2]: https://developer.arm.com/documentation/ddi0488/d/level-1-memory-system/l1-data-memory-system/preload-instruction-behavior
[3]: https://developer.arm.com/documentation/ddi0535/b/l1-memory-system/data-prefetching/pld--pldw--and-pli-instructions
[4]:  https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf 4-402 Vol. 2B
[5]: https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf 4-404 Vol. 2B